crates/nri-resctrl-plugin/src/lib.rs

```diff
@@ -0,0 +1,343 @@
+        debug!("resctrl-plugin: state_change: event={:?}", req.event);
+        Ok(Empty::default())
+    }
+
+    async fn shutdown(&self, _ctx: &TtrpcContext, _req: Empty) -> ttrpc::Result<Empty> {
+        info!("Shutting down resctrl plugin");
+        Ok(Empty::default())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use protobuf::SpecialFields;
+
+    #[test]
+    fn test_default_config() {
+        let cfg = ResctrlPluginConfig::default();
+        assert_eq!(cfg.group_prefix, "pod_");
+        assert!(cfg.cleanup_on_start);
+        assert_eq!(cfg.max_reconcile_passes, 10);
+        assert_eq!(cfg.concurrency_limit, 1);
+        assert_eq!(cfg.event_channel_capacity, 128);
+        assert!(!cfg.auto_mount);
+    }
+
+    #[tokio::test]
+    async fn test_configure_event_mask() {
+        let (plugin, _rx) = ResctrlPlugin::new(ResctrlPluginConfig::default());
+
+        let ctx = TtrpcContext {
+            mh: ttrpc::MessageHeader::default(),
+            metadata: std::collections::HashMap::<String, Vec<String>>::default(),
+            timeout_nano: 5_000,
+        };
+        let req = ConfigureRequest {
+            config: String::new(),
+            runtime_name: "test-runtime".into(),
+            runtime_version: "1.0".into(),
+            registration_timeout: 1000,
+            request_timeout: 1000,
+            special_fields: SpecialFields::default(),
+        };
+
+        let resp = plugin.configure(&ctx, req).await.unwrap();
+        let events = EventMask::from_raw(resp.events);
+
+        // Must include container create/stop and pod events
+        assert!(events.is_set(Event::CREATE_CONTAINER));
+        assert!(events.is_set(Event::STOP_CONTAINER));
+        assert!(events.is_set(Event::UPDATE_CONTAINER));
+        assert!(events.is_set(Event::UPDATE_POD_SANDBOX));
+        assert!(events.is_set(Event::RUN_POD_SANDBOX));
+        assert!(events.is_set(Event::STOP_POD_SANDBOX));
+        assert!(events.is_set(Event::REMOVE_POD_SANDBOX));
+    }
+
+    #[test]
+    fn test_channel_capacity_no_drops_under_normal_flow() {
+        let mut cfg = ResctrlPluginConfig::default();
+        cfg.event_channel_capacity = 4;
+        let (plugin, mut rx) = ResctrlPlugin::new(cfg);
+
+        for i in 0..4 {
+            plugin.emit_event(PodResctrlEvent::Added(PodResctrlAdded {
+                pod_uid: format!("pod-{i}"),
+                group_path: Some(format!("/sys/fs/resctrl/pod_{i}")),
+                state: AssignmentState::Success,
+            }));
+        }
+
+        // No drops expected
+        assert_eq!(plugin.dropped_events(), 0);
+
+        // Receive all 4
+        for _ in 0..4 {
+            let ev = rx.try_recv().expect("must receive event");
+            match ev {
+                PodResctrlEvent::Added(a) => {
+                    assert!(a.group_path.is_some());
+                }
+                _ => panic!("unexpected event variant"),
+            }
+        }
+    }
+}
```

Comment:
I'm missing an end-to-end test that shows that registration works with containerd's NRI infra. Can we create a tests/integration_test.rs that runs this? And we can add a job to the test-resctrl.yaml workflow that installs kind (with e.g., https://github.com/helm/kind-action) and runs the integration_test.rs inside the cluster's node, verifies that the test passes. the test would register the plugin, wait 100 milliseconds and close. that will give us confidence our registration and shutdown are ok, and will be a foundation for integration tests as we increase functionality